### PR TITLE
fix(zk): avoid observing log from staticcalls

### PR DESCRIPTION
**Pull request by Karrq**
## Motivation
After investigating #546 we noticed that logs were being processed also for `STATICCALL`, which aren't supposed to edit the state.

## Solution
Logs are part of the state, so we should ignore observing them for `STATICCALL`

### CODE REVIEW

File: crates/cheatcodes/src/inspector.rs [hunk 1]
```diff
@@ -1547,6 +1550,7 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
     }
 
     fn log(&mut self, _interpreter: &mut Interpreter, _context: &mut EvmContext<DB>, log: &Log) {
+        tracing::error!(?log, "log call");
```
Comment (nbaztec) [inline crates/cheatcodes/src/inspector.rs:1553]: why do we need this?

Comment (Karrq) [reply crates/cheatcodes/src/inspector.rs:1553]: we don't, I forgot to clean it up

File: testdata/zk/Cheatcodes.t.sol [hunk 2]
```diff
@@ -157,6 +164,14 @@ contract ZkCheatcodesTest is DSTest {
         new Emitter();
     }
 
+    function testExpectEmitIgnoresStaticCalls() public {
```
Comment (elfedy) [inline testdata/zk/Cheatcodes.t.sol:167]: I don't understand this test. Where is the static call happening and how is expect emit ignoring it?

Comment (Karrq) [reply testdata/zk/Cheatcodes.t.sol:167]: `emitter.FUNCTION_MESSAGE()`, due to it being a public field, would be the static call.
In general, view functions are static calls - perhaps it should be made clearer if `FUNCTION_MESSAGE` wasn't a constant, so the compiler couldn't optimize away the read

**Review by nbaztec — APPROVED** [review]

**Pull request merged**